website/src/components/subscriptions/overview/SubscriptionEntry.tsx

```diff
@@ -197,3 +210,42 @@ function DeleteButton({ onClick }: { onClick: JSX.IntrinsicElements['button']['o
         </button>
     );
 }
+
+function ConfirmDeletionModal({
+    onDelete,
+    modalRef,
+}: {
+    onDelete: JSX.IntrinsicElements['button']['onClick'];
+    modalRef: React.RefObject<HTMLDialogElement>;
```

Comment:
```suggestion
    modalRef: RefObject<HTMLDialogElement>;
```

and `import { type RefObject } from 'react';`